### PR TITLE
luci-app-openvpn: Implement autocomplete Name field based on the uploaded file name

### PR DESCRIPTION
**Pull request by digitaltechconsult**
luci-app-openvpn: Implement autocomplete Name field based on the uploaded file name.
Signed-off-by: Bogdan Coticopol

### CODE REVIEW

File: applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm [hunk 1]
```diff
@@ -67,6 +67,18 @@
 		{
 			setTimeout(function(){ output.innerHTML=""}, timeout);
 		}
+	}
```
Comment (feckert) [inline applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm:70]: Are you sure that the indentation is correct? The one with the bracket looks strange

File: applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm [hunk 2]
```diff
@@ -67,6 +67,18 @@
...
+		var vpn_file = document.getElementById("ovpn_file").value;
+
+		if(!vpn_name && !vpn_name.length) {
+			let vpn_file_name = vpn_file.replace('C:\\fakepath\\','').replace('.ovpn','').replace(/[^\x00-\x7F]|[\s\.!@#$%^&*()\-+=\[\]{};':"\\|,<>\/?]/g,'');
```
Comment (feckert) [inline applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm:74]: I find the regex difficult to understand. Isn't there something simple here?
Maybe a path split see https://www.geeksforgeeks.org/how-to-get-the-file-name-from-full-path-using-javascript/

I also think that the assumption that it is about **windows** is not correct either.

Comment (eraser2021999) [reply applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm:74]: The regex was already in place and it checks the name for illegal characters. You can check the code where the name is setup manually.

**Comment by systemcrash** [comment]
ping @eraser2021999

**Comment by Neustradamus** [comment]
I have sent an email to Bogdan Coticopol about this PR.

File: applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm (hunk 2 above)
Comment (systemcrash) [reply applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm:74]: https://developer.mozilla.org/en-US/docs/Web/API/FileList

Then avoid all the regex and just get the pure file name.

**Pull request closed without merging**